client/src/components/pickWinnerComponents/PickWinner.js

```diff
@@ -0,0 +1,70 @@
+import React, { Fragment, useContext, useEffect } from 'react';
+import {
+  Typography,
+  Container,
+  Box,
+  GridList,
+  GridListTile,
+  Button,
+} from '@material-ui/core';
+import { Redirect, useParams } from 'react-router-dom';
+import useStyles from './PickWinnerStyles';
+import UserContext from '../../context/UserContext';
+import { ContestContext } from '../../context/ContestContext';
+import IsLoading from '../IsLoading';
+
+const PickWinner = () => {
+  const classes = useStyles();
+  const params = useParams();
+  const userContext = useContext(UserContext);
+  const contestContext = useContext(ContestContext);
+
+  useEffect(() => {
+    contestContext.getContestById(params.id);
+  }, [params.id]);
+
+  const { user } = userContext.state;
+  const { contest } = contestContext.state;
+
+  return !contest.user ? (
+    IsLoading()
+  ) : user._id !== contest.user._id ? (
+    <Redirect to="/" />
+  ) : (
+    <Fragment>
+      <Container maxWidth="md">
+        <Box boxShadow={3} className={classes.mainBox}>
+          <Typography variant="h5" m={2} className={classes.title}>
+            Pick your winner, {contest.user.name}!
+          </Typography>
+          <Container className={classes.innerContainer}>
+            {contest.submissions.map((sub, i) => {
+              return (
+                <Box mb={5} key={i}>
```

Comment:
Is there anything unique on submissions that you can use as the key instead of the index?

Reply:
Ok. Fixed and will pay more attention to the key from now on. Is this emphasis on key names for accessibility reasons?

Reply:
It's actually for React to be able to intelligently decide when to re-render components. An index might change more often than something unique like an id, which could lead to unnecessary re-rendering. It doesn't always make a difference, but it's a good habit to have. https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
Cool. Thanks for the explanation.